app/serializers/queued_post_serializer.rb

```diff
@@ -1,36 +1,74 @@
+# Deprecated, should be removed once users have sufficient opportunity to do so
 class QueuedPostSerializer < ApplicationSerializer
 
-  attributes :id,
-             :queue,
-             :user_id,
-             :state,
-             :topic_id,
-             :approved_by_id,
-             :rejected_by_id,
-             :raw,
-             :post_options,
-             :created_at,
-             :category_id,
-             :can_delete_user
-
-  has_one :user, serializer: AdminUserListSerializer
+  attributes(
+    :id,
+    :queue,
+    :user_id,
+    :state,
+    :topic_id,
+    :approved_by_id,
+    :rejected_by_id,
+    :raw,
+    :post_options,
+    :created_at,
+    :category_id,
+    :can_delete_user
+  )
+  has_one :created_by, serializer: AdminUserListSerializer, root: :users
   has_one :topic, serializer: BasicTopicSerializer
 
-  def category_id
-    cat_id = object.topic.try(:category_id) || object.post_options['category']
-    cat_id.to_i if cat_id
+  def queue
+    'default'
   end
 
-  def include_category_id?
-    category_id.present?
+  def user_id
+    object.created_by_id
+  end
+
+  def state
+    object.status + 1
+  end
+
+  def approved_by_id
+    who_did(:approved)
+  end
+
+  def rejected_by_id
+    who_did(:rejected)
+  end
+
+  def raw
+    object.payload['raw']
+  end
+
+  def post_options
+    json = object.payload.dup
+    json.delete('raw')
+    json
```

Comment:
You can use `Hash#except` in the future :)

Reply:
nice tip, following up with a fix.